background_jobs/jobs/clear_blocks.go

```diff
@@ -0,0 +1,48 @@
+package jobs
+
+import (
+	"fmt"
+
+	"github.com/bitcoin-sv/arc/dbconn"
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
+
+var logger zerolog.Logger
+
+func init() {
+	zerolog.TimeFieldFormat = zerolog.TimeFormatUnix
+	logger = log.With().Str("job", "ClearBlocks").Logger()
+}
+
+type ClearBlockParams struct {
+	dbconn.DBConnectionParams
+	BlockRetentionDays int //delete blocks that are BlockRetentionDays days old
+}
+
+func ClearBlocks(params ClearBlockParams) error {
+	logger.Info().Msg("Connecting to database ...")
+	conn, err := sqlx.Open(params.Scheme, params.String())
+	if err != nil {
+		logger.Err(err).Msg("unable to create connection")
+		return err
+	}
+	interval := fmt.Sprintf("%d days", params.BlockRetentionDays)
+
+	stmt, err := conn.Preparex("DELETE FROM blocks WHERE inserted_at <= (CURRENT_DATE - $1::interval)")
```

Comment:
We need not only to delete from `blocks` but also the other tables. That is `transactions` and `block_transactions_map`

This query may be very slow, because there is no index on `inserted_at`

Reply:
I would not try to optimise queries until we really have bottleneck at this particular point.  Also will have to add inserted_at for those other fields.